Library/Homebrew/extend/os/mac/diagnostic.rb

```diff
@@ -438,6 +439,43 @@ def check_if_supported_sdk_available
           #{update_instructions}
         EOS
       end
+
+      # The CLT 10.x -> 11.x upgrade process on 10.14 contained a bug which broke the SDKs.
+      # Notably, MacOSX10.14.sdk would indirectly symlink to MacOSX10.15.sdk.
+      # This diagnostic was introduced to check for this and recommend a full reinstall.
+      def check_broken_sdks
+        locator = MacOS.sdk_locator
+
+        return if locator.all_sdks.all? do |sdk|
+          path_version = sdk.path.basename.to_s[MacOS::SDK::VERSIONED_SDK_REGEX, 1]
+          next true if path_version.blank?
+
+          path_version = MacOS::Version.new(path_version)
+
+          # Compare, excluding patch version
+          sdk.version == MacOS::Version.from_symbol(path_version.to_sym)
```

Comment:
You're converting to and from `MacOS::Version` symbols in two places here... Maybe it'd be better to just store `sdk.version` as a symbol?

Reply:
From an API standpoint, `MacOS::Version` is better and more flexible (and won't have the potential to break anything already using `MacOS.sdk.version`). We could have a `strip_patch` (or a better named) method instead of the conversion stuff though.

Reply:
Does `path_version.major_minor` not work?

Reply:
That returns a `::Version` - not a `MacOS::Version`. The latter is preferable so we can do `version >= :big_sur` rather than `version.major >= 11`.